src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,306 @@
+            if (!unprocessedHistoryEntries.Any())
+            {
+                Logger.LogInformation("No errors found in the RepositoryBranchUpdateHistory table.");
+                return;
+            }
+            Logger.LogInformation("Going to create the github issue.");
+            try
+            {
+                foreach (var error in unprocessedHistoryEntries)
+                {
+                    await CreateOrUpdateGithubIssueAsync(error, issueRepo);
+                    IReliableDictionary<string, DateTime> update =
+                                   await _stateManager.GetOrAddAsync<IReliableDictionary<string, DateTime>>("update");
+                    try
+                    {
+                        using (ITransaction tx = _stateManager.CreateTransaction())
+                        {
+                            await update.SetAsync(
+                                tx,
+                                "update",
+                                error.Timestamp
+                                );
+                            await tx.CommitAsync();
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogError(ex.Message, "Failed to retrieve previous transaction time.");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex.Message, "Could not create a github issue.");
+            }
+        }
+
+        private async Task<GitHubClient> AuthenticateGitHubClient(string issueRepo)
+        {
+            string gitHubToken = null;
+            IGitHubTokenProvider gitHubTokenProvider = Context.GetService<IGitHubTokenProvider>();
+            long installationId = await Context.GetInstallationId(issueRepo);
+            gitHubToken = await gitHubTokenProvider.GetTokenForInstallationAsync(installationId);
+            Logger.LogInformation("GitHub token acquired for " + issueRepo);
+            Octokit.ProductHeaderValue _product;
+            string version = Assembly.GetExecutingAssembly()
+                .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                .InformationalVersion;
+            _product = new Octokit.ProductHeaderValue("Maestro", version);
+            var client = new GitHubClient(_product);
+            var token = new Credentials(gitHubToken);
+            client.Credentials = token;
+            return client;
+        }
+
+        private async Task CreateOrUpdateGithubIssueAsync(RepositoryBranchUpdateHistoryEntry repositoryBranchUpdateHistory , string issueRepo)
+        {
+            Logger.LogInformation("Something failed in the repository : " + repositoryBranchUpdateHistory.Repository);
+            string fyiHandles = _options.FyiHandle;
+            IReliableDictionary<(string, string), int> gitHubIssueEvaluator =
```

Comment:
Can we name these tuples?  "string, string" isn't very helpful when it comes to reading the code.